### PR TITLE
Update API docs

### PR DESCRIPTION
**Pull request by jimmyjames**
### Changes

Updates the doc strings for the public APIs, to ensure the example usages are correct and demonstrated in Kotlin.

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [X] All existing and new tests complete without errors

### CODE REVIEW

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 1]
```diff
@@ -46,11 +46,10 @@ import java.security.PublicKey
 
 /**
  * API client for Auth0 Authentication API.
- * <pre>
- * `Auth0 auth0 = new Auth0("your_client_id", "your_domain");
- * AuthenticationAPIClient client = new AuthenticationAPIClient(auth0);
-` *
-</pre> *
+ * ```
+ * val auth0 = Auth0("your_client_id", "your_domain")
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:50]: we probably should be consistent with the use of these "placeholders" and the casing

```suggestion
 * val auth0 = Auth0("YOUR_CLIENT_ID", "YOUR_DOMAIN")
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 2]
```diff
@@ -94,18 +93,14 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     *
-     * {}Override
-     * public void onFailure(AuthenticationException error) { }
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client
+     *     .login("{username or email}", "{password}", "{database connection name}")
+     *     .start(object : AuthenticationCallback<Credentials> {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:99]: I noticed you swapped the class to `AuthenticationCallback`. Do we want to make these changes now or in the coming weeks?

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 3]
```diff
@@ -379,17 +346,13 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(Credentials payload) {}
-     *
-     * {}Override
-     * public void onFailure(AuthenticationException error) {}
-     * });DefaultClient(auth0.connectTimeoutInSeconds)
-    ` *
-    </pre> *
+     * ```
+     * client.signUp("{email}", "{password}", "{username}", "{database connection name}")
+     *     .start(object: Callback<Credentials, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:351]: Here you are using Callback again. Let's be consistent on either Callback<T, U> or AuthenticationException<T>

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 4]
```diff
@@ -622,13 +565,12 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
      * Fetch the token information from Auth0, using the authorization_code grant type
      * The authorization code received from the Auth0 server and the code verifier used
      * to generate the challenge sent to the /authorize call must be provided.
-     * <pre>
-     * `AuthenticationAPIClient client = new AuthenticationAPIClient(new Auth0("clientId", "domain"));
+     * ```
+     * val client = AuthenticationAPIClient(Auth0("clientId", "domain"))
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:569]: I noticed this one shows how to instantiate the client. We probably don't want this here, but maybe is useful to keep in the class constructor?

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt (hunk 1 above)
Comment (jimmyjames) [reply auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt]: New commit to address.

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt (hunk 2 above)
Comment (jimmyjames) [reply auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt]: New commit to use `Callback` in docs.

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt (hunk 3 above)
Comment (jimmyjames) [reply auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:355]: New commit to use `Callback` in docs.

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt (hunk 4 above)
Comment (jimmyjames) [reply auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt]: New commit to remove instantiation example here, and add to constructor.

**Review by lbalmaceda — APPROVED** [review]

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 5]
```diff
@@ -189,17 +180,14 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(Credentials payload) { }
      *
-     * {}Override
-     * public void onFailure(AuthenticationException error) { }
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.loginWithNativeSocialToken("{subject token}", "{subject token type}")
+     *     .start(object: Callback<Credentials, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:186]: I think this doesn't take the `()`
image

```suggestion
     *     .start(object: Callback<Credentials, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 6]
```diff
@@ -271,18 +253,15 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
      *
-     * {}@Override
-     * public void onFailure(AuthenticationException error) { }
-     * });
-    ` *
-    </pre> *
+     * Example usage:
+     * ```
+     * client.loginWithEmail("{email}", "{code}", "{passwordless connection name}")
+     *     .start(object: Callback<Credentials, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:260]:
```suggestion
     *     .start(object: Callback<Credentials, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 7]
```diff
@@ -228,22 +216,16 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(Credentials payload) { }
-     *
-     * {}@Override
-     * public void onFailure(AuthenticationException error) { }
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.loginWithPhoneNumber("{phone number}", "{code}", "{passwordless connection name}")
+     *     .start(object: Callback<Credentials, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:224]:
```suggestion
     *     .start(object: Callback<Credentials, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 8]
```diff
@@ -309,17 +288,13 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(UserProfile payload) { }
-     *
-     * {}@Override
-     * public void onFailure(AuthenticationException error) { }
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.userInfo("{access_token}")
+     *     .start(object: Callback<UserProfile, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:293]:
```suggestion
     *     .start(object: Callback<UserProfile, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 9]
```diff
@@ -332,17 +307,13 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(DatabaseUser payload) { }
-     *
-     * {}@Override
-     * public void onFailure(AuthenticationException error) { }
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.createUser("{email}", "{password}", "{username}", "{database connection name}")
+     *     .start(object: Callback<DatabaseUser, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:312]:
```suggestion
     *     .start(object: Callback<DatabaseUser, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 10]
```diff
@@ -379,17 +350,13 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(Credentials payload) {}
-     *
-     * {}Override
-     * public void onFailure(AuthenticationException error) {}
-     * });DefaultClient(auth0.connectTimeoutInSeconds)
-    ` *
-    </pre> *
+     * ```
+     * client.signUp("{email}", "{password}", "{username}", "{database connection name}")
+     *     .start(object: Callback<Credentials, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:355]:
```suggestion
     *     .start(object: Callback<Credentials, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 11]
```diff
@@ -412,17 +379,13 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(Void payload) {}
-     *
-     * {}Override
-     * public void onFailure(AuthenticationException error) {}
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.resetPassword("{email}", "{database connection name}")
+     *     .start(object: Callback<Void, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:384]:
```suggestion
     *     .start(object: Callback<Void, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 12]
```diff
@@ -449,17 +412,13 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(Void payload) {}
-     *
-     * {}Override
-     * public void onFailure(AuthenticationException error) {}
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.revokeToken("{refresh_token}")
+     *     .start(object: Callback<Void, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:417]:
```suggestion
     *     .start(object: Callback<Void, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 13]
```diff
@@ -484,18 +443,14 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     *
-     * {}@Override
-     * public void onFailure(AuthenticationException error) { }
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.renewAuth("{refresh_token}")
+     *     .addParameter("scope", "openid profile email")
+     *     .start(object: Callback<Credentials, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:449]:
```suggestion
     *     .start(object: Callback<Credentials, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 14]
```diff
@@ -521,17 +476,13 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(Void payload) {}
-     *
-     * {}Override
-     * public void onFailure(AuthenticationException error) {}
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.passwordlessWithEmail("{email}", PasswordlessType.CODE, "{passwordless connection name}")
+     *     .start(object: Callback<Void, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:481]:
```suggestion
     *     .start(object: Callback<Void, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt [hunk 15]
```diff
@@ -557,17 +508,13 @@ public class AuthenticationAPIClient @VisibleForTesting(otherwise = VisibleForTe
...
-     * public void onSuccess(Void payload) {}
-     *
-     * {}Override
-     * public void onFailure(AuthenticationException error) {}
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.passwordlessWithSms("{phone number}", PasswordlessType.CODE, "{passwordless connection name}")
+     *     .start(object: Callback<Void, AuthenticationException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/authentication/AuthenticationAPIClient.kt:513]:
```suggestion
     *     .start(object: Callback<Void, AuthenticationException> {
```

File: auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt [hunk 16]
```diff
@@ -90,17 +89,13 @@ public class UsersAPIClient @VisibleForTesting(otherwise = VisibleForTesting.PRI
...
-     * public void onSuccess(List<UserIdentity> payload) {}
-     *
-     * {}Override
-     * public void onFailure(ManagementException error) {}
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.link("{auth0 primary user id}", "{user secondary token}")
+     *     .start(object: Callback<List<UserIdentity>, ManagementException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt:94]:
```suggestion
     *     .start(object: Callback<List<UserIdentity>, ManagementException> {
```

File: auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt [hunk 17]
```diff
@@ -130,17 +125,13 @@ public class UsersAPIClient @VisibleForTesting(otherwise = VisibleForTesting.PRI
...
-     * public void onSuccess(List<UserIdentity> payload) {}
-     *
-     * {}Override
-     * public void onFailure(ManagementException error) {}
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.unlink("{auth0 primary user id}", {auth0 secondary user id}, "{secondary provider}")
+     *     .start(object: Callback<List<UserIdentity>, ManagementException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt:130]:
```suggestion
     *     .start(object: Callback<List<UserIdentity>, ManagementException> {
```

File: auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt [hunk 18]
```diff
@@ -170,17 +161,13 @@ public class UsersAPIClient @VisibleForTesting(otherwise = VisibleForTesting.PRI
...
-     * public void onSuccess(UserProfile payload) {}
-     *
-     * {}Override
-     * public void onFailure(ManagementException error) {}
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.updateMetadata("{user id}", "{user metadata}")
+     *     .start(object: Callback<UserProfile, ManagementException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt:166]:
```suggestion
     *     .start(object: Callback<UserProfile, ManagementException> {
```

File: auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt [hunk 19]
```diff
@@ -206,17 +193,13 @@ public class UsersAPIClient @VisibleForTesting(otherwise = VisibleForTesting.PRI
...
-     * public void onSuccess(UserProfile payload) {}
-     *
-     * {}Override
-     * public void onFailure(ManagementException error) {}
-     * });
-    ` *
-    </pre> *
+     * ```
+     * client.getProfile("{user id}")
+     *     .start(object: Callback<UserProfile, ManagementException>() {
```
Comment (lbalmaceda) [inline auth0/src/main/java/com/auth0/android/management/UsersAPIClient.kt:198]:
```suggestion
     *     .start(object: Callback<UserProfile, ManagementException> {
```

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

**Pull request merged**